src/long-to-linked-pe.cpp

```diff
@@ -0,0 +1,318 @@
+				} else {
+					bx_multiplicity_ofs << record.name << "\t" << seq_size / step << std::endl;
+				}
+			}
+			if (!with_bx_multiplicity_only) {
+
+				std::string& qual = record.qual;
+
+				size_t qual_size = qual.size();
+
+				if (auto_dist) {
+					if (seq_size > dist_lower_bound) {
+						read_lengths.push_back(seq_size);
+					}
+				}
+				if (auto_span) {
+					total_bases += seq_size;
+				}
+
+				int read_num = 1;
+				for (size_t i = 0; i <= seq_size - step; i = i + step) {
+					std::cout << header_symbol << record.name << "_f" << read_num
+					          << " BX:Z:" << record.num + 1 << '\n';
+					std::cout << seq.substr(i, l) << '\n';
+					if (!with_fasta) {
+						std::cout << "+\n";
+						if (qual_size == 0) {
+							std::cout << std::string(l, '#') << '\n';
+						} else {
+							std::cout << qual.substr(i, l) << '\n';
+						}
+					}
+					std::string reverse_linked_read = seq.substr(i + l, l);
+					btllib::reverse_complement(reverse_linked_read);
+					std::cout << header_symbol << record.name << "_f" << read_num
+					          << " BX:Z:" << record.num + 1 << '\n';
+					std::cout << reverse_linked_read << '\n';
+					if (!with_fasta) {
+						std::cout << "+\n";
+						if (qual_size == 0) {
+							std::cout << std::string(l, '#') << '\n';
+						} else {
+							std::string reverse_qual = qual.substr(i + l, l);
+							std::reverse(reverse_qual.begin(), reverse_qual.end());
+							std::cout << reverse_qual << '\n';
+						}
+					}
+					++read_num;
+				}
+
+				size_t remainder = seq_size % step;
+				if (remainder != 0) {
+					size_t curr_i = seq_size - remainder;
+					std::cout << header_symbol << record.name << "_f" << read_num
+					          << " BX:Z:" << record.num + 1 << '\n';
+					std::string forward_linked_read = seq.substr(curr_i, l);
+					std::cout << forward_linked_read << '\n';
+					if (!with_fasta) {
+						std::cout << "+\n";
+						if (qual_size == 0) {
+							std::cout << std::string(forward_linked_read.size(), '#') << '\n';
+						} else {
+							std::cout << qual.substr(curr_i, l) << '\n';
+						}
+					}
+					std::string reverse_linked_read = seq.substr(
+					    seq_size - forward_linked_read.size(), forward_linked_read.size());
+					btllib::reverse_complement(reverse_linked_read);
+					std::cout << header_symbol << record.name << "_f" << read_num
+					          << " BX:Z:" << record.num + 1 << '\n';
+					std::cout << reverse_linked_read << '\n';
+					if (!with_fasta) {
+						std::cout << "+\n";
+						if (qual_size == 0) {
+							std::cout << std::string(forward_linked_read.size(), '#') << '\n';
+						} else {
+							std::string reverse_qual = qual.substr(
+							    seq_size - forward_linked_read.size(), forward_linked_read.size());
+							std::reverse(reverse_qual.begin(), reverse_qual.end());
+							std::cout << reverse_qual << '\n';
+						}
+					}
+				}
+
+				std::cout << std::flush;
+			}
+		}
+	}
+
+	if (auto_span || auto_dist) {

```

Comment:
Just checking - this writing will _append_ if needed, right? Ie. I may write the `dist` to the file, and then run this to add the `span`

Reply:
everything is now appended